src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1693,6 +1693,7 @@ private void SetPeriodicEvaluation(WorkspaceModel ws)
           bool forceManualExecutionMode,
           out WorkspaceModel workspace)
         {
+            CheckForIronPythonDependencies(fileContents);
```

Comment:
So this will happen before Dynamo tries to open the file.. 
1. About the behavior. This may be less ideal than after the graph has been opened so people can see the graph, then the dialog.. I guess we did not make it clear @mjkkirschner @Amoursol .
2. Also curious about the performance impact on this as well since we can do the check in the `FromJson` function below as well to reduce one time of Json deserialization

Reply:
I don't think this needs to be done here - As @QilongTang said theres a few reasons.

1. would be nice if users could see the graph when the warning displays, instead of the start page.

2. The entire file is being parsed by json.net a second time, I think you can just let the graph get opened, and then check the deserialized nodes later using the engine property, instead of checking the JSON. It's better to rely on the runtime apis, even internal ones I think than to use the serialized json via string comparison.

Reply:
on second thought - I guess one reason you went with this approach was to avoid needing to reference PythonNodeModel from DynamoCore - and this is a good thing. We do not want to have a compile time or runtime dependency between them.

I would ask that you consider alternatives to the JSON parsing with the constraint mentioned above and see if you can think of another approach. If possible I would also avoid reflection if you need to scan everynode as that will be slow.

⚠️ 
Here's another idea...
Why don't you just have the PythonNodeModel constructor raise this notification itself, theres no reason to need to scan anything.

Reply:
@mjkkirschner yep, @SHKnudsen was trying to avoiding that circular dependency.

If we're ok with raising this `after` the workspace has loaded, then we could look to move all of this into an extension, should give us the required event hooks and can safely reference `PythonNodeModels` project too, let us know if there's any objections to going that route instead ?

Reply:
@radumg @SHKnudsen @QilongTang  - hmm - heres some thoughts.
* we know we're eventually going to want to create a migration assistant, it seems likely it will need some application level access - what if we start this now by creating a new extension like @radumg suggests.
* we can use that extension to unify both the warning dialog and the warning notifications on the python nodes.
* The extension can reference python without adding forcing a reference on python to any of our core extensions or binaries.
* We need to design that extension in such a way that it is robust and does not crash, even if the python or IronPython binaries are completely missing, as this may be the case in the future.

still thinking through this for now.

Reply:
just to be clear - I was not imagining this new extension to have any UI at this time - but only act as a centralized location for python migration analysis, raising of warnings and notifications, etc.

Reply:
@mjkkirschner @QilongTang 

added the extension to handle the node and graph warnings, along with a few other small changes